service/src/main/scala/org/apache/celeborn/server/common/HttpService.scala

```diff
@@ -27,15 +33,107 @@ abstract class HttpService extends Service with Logging {
   def getConf: String = {
     val sb = new StringBuilder
     sb.append("=========================== Configuration ============================\n")
-    if (conf.getAll.length > 0) {
+    if (conf.getAll.nonEmpty) {
       val maxKeyLength = conf.getAll.toMap.keys.map(_.length).max
-      conf.getAll.foreach { case (key, value) =>
-        sb.append(s"${key.padTo(maxKeyLength + 10, " ").mkString}$value\n")
+      conf.getAll.sortBy(_._1).foreach { case (key, value) =>
+        sb.append(config(key, value, maxKeyLength))
       }
     }
     sb.toString()
   }
 
+  def getDynamicConfig(
+      level: String,
+      tenant: String,
+      name: String): String = {
+    if (configService == null) {
+      s"Dynamic configuration is disabled. Please check whether to config `${CelebornConf.DYNAMIC_CONFIG_STORE_BACKEND.key}`."
+    } else {
+      val sb = new StringBuilder
+      sb.append("=========================== Dynamic Configuration ============================\n")
+      level.split(",").filter(_.nonEmpty).foreach { configLevel =>
+        sb.append(
+          s"=========================== Level: $configLevel ============================\n")
+        if (ConfigLevel.SYSTEM.name().equals(configLevel)) {
+          sb.append(systemConfigs())
+        } else if (ConfigLevel.TENANT.name().equals(configLevel)) {
+          sb.append(tenantConfigs(tenant))
+        } else if (ConfigLevel.TENANT_USER.name().equals(configLevel)) {
+          sb.append(tenantUserConfigs(tenant, name))
+        }
+      }
+      sb.toString()
+    }
+  }
+
+  private def systemConfigs(): String = {
+    // When setting config level is SYSTEM, appends all system level configs.
+    configs(configService.getSystemConfigFromCache.getConfigs)
+  }
+
+  private def tenantConfigs(tenant: String): String = {
+    // When setting config level is TENANT without tenant ids, appends all tenant level configs.
+    // When setting config level is TENANT with tenant ids, appends only tenant level configs of given tenant ids.
+    val tenantConfigs =
+      if (tenant.isEmpty) {
+        configService.listRawTenantConfigsFromCache().asScala
+      } else {
+        tenant.split(",").map { tenantId =>
+          configService.getRawTenantConfigFromCache(tenantId)
+        }.toList
+      }
+    tenantConfigs.sortBy(_.getTenantId).map { tenantConfig =>
+      s"""
+         |=========================== Tenant: ${tenantConfig.getTenantId} ============================
+         |${configs(tenantConfig.getConfigs)}""".stripMargin
+    }.mkString("\n")
+  }
+
+  private def tenantUserConfigs(tenant: String, name: String): String = {
```

Comment:
IMO and from the parameter of tenantUserConfigs, we should only support single tenant/single tenant with username to simplify and avoid the combinations of tenantId and usernames, if tenant/name are both empty we list all configs by config_level